packages/x-adapter/src/http-clients/fetch.http-client.ts

```diff
@@ -18,8 +18,8 @@ export const fetchHttpClient: HttpClient = (
 ) => {
   const signal = cancelable ? { signal: abortAndGetNewAbortSignal(id) } : {};
   const flatParameters = flatObject(parameters);
-  const url = sendParamsInBody ? endpoint : buildUrl(endpoint, flatParameters);
-  const bodyParameters = sendParamsInBody ? { body: JSON.stringify(parameters) } : {};
+  const url = sendParamsInBody ? endpoint : buildUrl(endpoint, cleanEmpty(flatParameters));
+  const bodyParameters = sendParamsInBody ? { body: JSON.stringify(cleanEmpty(parameters)) } : {};
```

Comment:
@LuisMartinez15 suggested maybe doing the clean of the parameters first, so `cleanEmpty` is called just once: 
```
const cleanedParameters = cleanEmpty(parameters);
const flatParameters = flatObject(cleanedParameters);
const url = sendParamsInBody ? endpoint : buildUrl(endpoint, flatParameters);
const bodyParameters = sendParamsInBody ? { body: JSON.stringify(cleanedParameters) } : {};
```
In this case, we would have a different result for objects like this:
```
{
   q: 'hey',
   whatever: 'hola',
   extraParams: {
      whatever: ''
   }
}
```
In one case, we will have in `flatParameters` an object containing `whatever: 'hola'`, and in the other, `whatever` would be completely removed. 

@javieri-empathy @CachedaCodes which one do you think is the desired behavior?

Reply:
Hmm good catch!. After thinking about this and talking with you, I would leave it as it is (1st flat -> 2nd clean). If you use extra params to override a harcoded param of the request you have to accept the full consequences. If you provide an empty extra param property value it should remove the default parameter.

Checking this, we spotted that `flatObject` has a weird behavior. Depending on how you create the object, the order the keys are inserted, it will return different results. I have created a [task to check this.](https://searchbroker.atlassian.net/browse/EX-7019)